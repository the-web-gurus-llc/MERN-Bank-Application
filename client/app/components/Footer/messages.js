/*
 * Footer Messages
 *
 * This contains all the text for the Footer component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.Footer';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Remember the basic safety rules.',
  },
  subheader: {
    id: `${scope}.subheader`,
    defaultMessage:
      'Before you enter your ID number and access password on your website, make sure',
  },
  ul_li1: {
    id: `${scope}.ul_li1`,
    defaultMessage:
      'Your password is secure. It contains at least 8 characters and consists of uppercase and lowercase letters',
  },
  ul_li2: {
    id: `${scope}.ul_li2`,
    defaultMessage:
      'in the address bar or status bar, a padlock is visible at the bottom of the browser screen',
  },
  warning: {
    id: `${scope}.warning`,
    defaultMessage:
      'Remember: The bank does not require confirmation of any data, correct login or reading of the Banks messages by means of SMS, TAN and / or e-mail, or installation of any applications on the users phones or computers.',
  },
  footer: {
    id: `${scope}.footer`,
    defaultMessage:
      'This site uses cookies to provide services at the highest level. By clicking or navigating the site, you agree to allow our collection of information on through cookies. For more information on security, please visit:',
  },
  buttonContent: {
    id: `${scope}.buttonContent`,
    defaultMessage: 'Privacy rules',
  },
});
