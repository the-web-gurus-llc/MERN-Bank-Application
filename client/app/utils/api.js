const API_BASE_URL = 'https://api.pietrzakadrian.com/bank';
const AUTH = 'Auth';
const USERS = 'Users';
const CURRENCIES = 'Currencies';
const BILLS = 'Bills';
const TRANSACTIONS = 'Transactions';
const MESSAGES = 'Messages';
const NOTIFICATIONS = 'Notifications';

export default {
  auth: {
    login: `${API_BASE_URL}/${AUTH}/login`,
    register: `${API_BASE_URL}/${AUTH}/register`,
    logout: `${API_BASE_URL}/${AUTH}/logout`,
    forgetPassword: `${API_BASE_URL}/${AUTH}/password/forget`,
    resetPassword: `${API_BASE_URL}/${AUTH}/password/reset`,
  },
  users: (path = '') => {
    switch (path) {
      case 'checkEmail':
        return (email) => `${API_BASE_URL}/${USERS}/${email}/checkEmail`;
      default:
        return `${API_BASE_URL}/${USERS}`;
    }
  },
  currencies: `${API_BASE_URL}/${CURRENCIES}`,
  messages: `${API_BASE_URL}/${MESSAGES}`,
  notifications: `${API_BASE_URL}/${NOTIFICATIONS}`,
  bills: (path = '') => {
    switch (path) {
      case 'amountMoney':
        return `${API_BASE_URL}/${BILLS}/amountMoney`;
      case 'accountBalance':
        return `${API_BASE_URL}/${BILLS}/accountBalance`;
      case 'accountBalanceHistory':
        return `${API_BASE_URL}/${BILLS}/accountBalanceHistory`;
      case 'search':
        return (accountBillNumber) =>
          `${API_BASE_URL}/${BILLS}/${accountBillNumber}/search`;
      default:
        return `${API_BASE_URL}/${BILLS}`;
    }
  },
  transactions: (path = '') => {
    switch (path) {
      case 'create':
        return `${API_BASE_URL}/${TRANSACTIONS}/create`;
      case 'confirm':
        return `${API_BASE_URL}/${TRANSACTIONS}/confirm`;
      case 'authorizationKey':
        return (uuid) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/authorizationKey`;
      case 'confirmationFile':
        return (uuid, locale) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/${locale}/confirmationFile`;
      default:
        return `${API_BASE_URL}/${TRANSACTIONS}`;
    }
  },
};
