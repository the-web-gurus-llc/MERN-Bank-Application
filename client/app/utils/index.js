export { default as colors } from './colors';
export { default as media } from './media';
export { default as request } from './request';
export { default as typography } from './typography';
export { default as api } from './api';
export { default as routes } from './routes';
export { default as locales } from './locales';
export { default as i18n } from './i18n';
