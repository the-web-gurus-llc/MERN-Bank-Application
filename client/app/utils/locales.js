exports.DEFAULT_LOCALE = 'en';
exports.DEFAULT_DATE_FORMAT = 'dd.MM.yyyy, hh:mm aa';

// prettier-ignore
exports.appLocales = [
  'en',
  'de',
  'pl',
];
