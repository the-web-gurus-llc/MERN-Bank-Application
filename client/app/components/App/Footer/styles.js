import styled from 'styled-components';
import { media } from 'utils';

export const StyledFooter = styled.footer`
  display: inline-block;
  color: black;
  text-align: center;
  position: absolute;
  width: 100%;
  left: 50%;
  transform: translate(-50%, 0);
  bottom: 0;
  padding: 10px;
  border-top: 1px solid rgba(0, 0, 0, 0.12);
  font-size: 12px;
  white-space: nowrap;
  transition: 0.4s;

  ${media.tablet`
    opacity: ${({ collapsed }) => (collapsed ? 0 : 1)}};
    visibility: ${({ collapsed }) => (collapsed ? 'hidden' : 'visible')}};
  `};
`;
