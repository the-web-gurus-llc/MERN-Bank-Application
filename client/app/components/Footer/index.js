/**
 *
 * Footer
 *
 */

import React from 'react';
import { useDispatch } from 'react-redux';
import { push } from 'connected-react-router';
import { FormattedMessage } from 'react-intl';
import { routes } from 'utils';
import {
  StyledFooter,
  StyledButton,
  StyledInfoCircleOutlined,
  StyledWarning,
  StyledTip,
} from './styles';
import messages from './messages';

export default function Footer() {
  const dispatch = useDispatch();

  return (
    <StyledFooter>
      <StyledWarning>
        <StyledInfoCircleOutlined />
        <FormattedMessage {...messages.header} />
      </StyledWarning>

      <div>
        <FormattedMessage {...messages.subheader} />
        <ul>
          <li>
            <FormattedMessage {...messages.ul_li1} />
          </li>
          <li>
            <FormattedMessage {...messages.ul_li2} />
          </li>
        </ul>
      </div>

      <StyledTip>
        <FormattedMessage {...messages.warning} />
      </StyledTip>

      <div>
        <FormattedMessage {...messages.footer} />{' '}
        <StyledButton
          type="link"
          onClick={() => dispatch(push(routes.privacy.path))}
        >
          <FormattedMessage {...messages.buttonContent} />
        </StyledButton>
      </div>
    </StyledFooter>
  );
}
