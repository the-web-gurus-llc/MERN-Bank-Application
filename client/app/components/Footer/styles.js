import styled from 'styled-components';
import { Button } from 'antd';
import { InfoCircleOutlined } from '@ant-design/icons';
import { typography, colors } from 'utils';

export const StyledFooter = styled.footer`
  margin: 15px auto;
  max-width: 560px;
  font-size: 13px;
  padding: 0 20px 10px;
`;

export const StyledWarning = styled.div`
  display: flex;
  align-items: center;
  font-weight: ${typography.fontWeightMedium};
  margin: 20px 0;
  font-size: 14px;
`;

export const StyledInfoCircleOutlined = styled(InfoCircleOutlined)`
  font-size: 40px;
  margin-right: 13px;
`;

export const StyledTip = styled.div`
  color: ${colors.red};
  margin: 15px 0;
`;

export const StyledButton = styled(Button)`
  font-weight: ${typography.fontWeightMedium};
  padding: 0;
  color: inherit;
  height: auto;
  font-size: inherit;
`;
