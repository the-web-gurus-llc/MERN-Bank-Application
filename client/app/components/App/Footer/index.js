/**
 *
 * Footer
 *
 */

import React from 'react';
import { createStructuredSelector } from 'reselect';
import { makeSelectIsCollapsedSidebar } from 'containers/App/selectors';
import { useSelector } from 'react-redux';
import { StyledFooter } from './styles';

const stateSelector = createStructuredSelector({
  isCollapsedSidebar: makeSelectIsCollapsedSidebar(),
});

export default function Footer() {
  const { isCollapsedSidebar } = useSelector(stateSelector);

  return (
    <StyledFooter collapsed={isCollapsedSidebar}>
      <strong>Bank Application 2.1</strong> | 19.12.2020
    </StyledFooter>
  );
}
