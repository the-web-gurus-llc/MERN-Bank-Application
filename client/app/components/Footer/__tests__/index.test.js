/**
 *
 * Tests for Footer
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';

import { createMemoryHistory } from 'history';
import { DEFAULT_LOCALE } from 'utils/locales';
import configureStore from 'utils/configureStore';
import Footer from '../index';

describe('<Footer />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render a div', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Footer />
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should handle click events', () => {
    const { getByText } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Footer />
        </IntlProvider>
      </Provider>,
    );

    const button = getByText('Privacy rules');
    fireEvent.click(button);
  });
});
