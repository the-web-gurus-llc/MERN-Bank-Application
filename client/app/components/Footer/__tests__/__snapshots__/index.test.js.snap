// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render a div 1`] = `
<footer
  class="Footerstyle__StyledFooter-sc-3vvbi3-0 efEtfj"
>
  <div
    class="Footerstyle__StyledWarning-sc-3vvbi3-1 btJlec"
  >
    <span
      aria-label="info-circle"
      class="anticon anticon-info-circle Footerstyle__StyledInfoCircleOutlined-sc-3vvbi3-2 iXYsXs"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="info-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
        />
        <path
          d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
        />
      </svg>
    </span>
    <span>
      Remember the basic safety rules.
    </span>
  </div>
  <div>
    <span>
      Before you enter your ID number and access password on your website, make sure
    </span>
    <ul>
      <li>
        <span>
          Your password is secure. It contains at least 8 characters and consists of uppercase and lowercase letters
        </span>
      </li>
      <li>
        <span>
          in the address bar or status bar, a padlock is visible at the bottom of the browser screen
        </span>
      </li>
    </ul>
  </div>
  <div
    class="Footerstyle__StyledTip-sc-3vvbi3-3 feybDB"
  >
    <span>
      Remember: The bank does not require confirmation of any data, correct login or reading of the Banks messages by means of SMS, TAN and / or e-mail, or installation of any applications on the users phones or computers.
    </span>
  </div>
  <div>
    <span>
      This site uses cookies to provide services at the highest level. By clicking or navigating the site, you agree to allow our collection of information on through cookies. For more information on security, please visit:
    </span>
     
    <button
      class="ant-btn Footerstyle__StyledButton-sc-3vvbi3-4 byxUJI ant-btn-link"
      type="button"
    >
      <span>
        Privacy rules
      </span>
    </button>
  </div>
</footer>
`;
